src/Lean/Compiler/LCNF/PassManager.lean

```diff
@@ -11,27 +11,118 @@ import Lean.Compiler.LCNF.CompilerM
 
 namespace Lean.Compiler.LCNF
 
+/--
+The pipeline phase a certain `Pass` is supposed to happen in.
+-/
+inductive Phase where
+| /-- Here we still carry most of the original type information, most
+      of the dependent portion is already (partially) erased though. -/
+  base
+| /-- Here all types have been turned into `Any`. -/
+  untyped
```

Comment:
In the second phase, the plan is to eliminate polymorphism, but we will preserve the types. We can call it `mono`.
For example, the declaration 
```lean
def f (α : Type u) (p : List (α × α)) : List (α × α)
```
becomes 
```lean
def f (α : ◾) (p : List (⊤ × ⊤) : List (⊤ × ⊤)
```
where ` ◾` represents erased information and `⊤` is the any type. 
The type of 
```lean
def g (as : List (Nat × Bool) : Nat
```
is not altered.